src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php

```diff
@@ -3676,6 +3714,28 @@ private static function getGreaterAssertions(
             return $if_types ? [$if_types] : [];
         }
 
+        if ($superior_value_position) {
+            if ($superior_value_position === self::ASSIGNMENT_TO_RIGHT) {
+                $var_name = ExpressionIdentifier::getArrayVarId(
+                    $conditional->left,
+                    $this_class_name,
+                    $source
+                );
+            } else {
+                $var_name = ExpressionIdentifier::getArrayVarId(
+                    $conditional->right,
+                    $this_class_name,
+                    $source
+                );
+            }
+
+            if ($var_name) {
+                $if_types[$var_name] = [['=isset'], ['+' . $superior_value_comparison]];
```

Comment:
Is there any reason why you chose `+`? I think this will eventually bleed out to user-defined assertions and `@psalm-assert +5 $param` is not very intuitive notion for a number that is at most 5.

Reply:
Damn, I didn't thought about user assertions. I didn't pick ">" because it's used here: https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/Statements/Expression/AssertionFinder.php#L96 and https://github.com/vimeo/psalm/blob/master/src/Psalm/Type/Reconciler.php#L165. I thought about using `>>` instead but assertions can be negated so ">" in second position is checked too somewhere.

We could replace ">" as it is already used by something else but I'm not sure to understand what's going on with that operator

Reply:
> We could replace ">" as it is already used by something else but I'm not sure to understand what's going on with that operator

Me neither.

Reply:
@muglug any insight here as to what's done with the ">" operator in assertions? The goal would be to change it to somewhere less significant so we can use ">" for superior

Reply:
I fixed it by replacing it with "@".